src/AlgebraicGeometry/ToricVarieties/NormalToricVarieties/constructors.jl

```diff
@@ -181,16 +181,50 @@ end
 # Equality
 ######################
 
-function Base.:(==)(tv1::NormalToricVariety, tv2::NormalToricVariety)
-  tv1 === tv2 && return true
-  error("Equality of normal toric varieties is computationally very demanding. More details in the documentation.")
-end
+@doc raw"""
+    (==)(X::NormalToricVariety, Y::NormalToricVariety)
 
-function Base.hash(tv::NormalToricVariety, h::UInt)
-  return hash(objectid(tv), h)
-end
+Checks equality of the polyhedral fans as sets of cones.
+
+# Examples
+```jldoctest
+julia> H = hirzebruch_surface(NormalToricVariety, 0)
+Normal toric variety
+
+julia> P1 = projective_space(NormalToricVariety, 1)
+Normal toric variety
+
+julia> H == P1 * P1
+true
+```
+"""
+function Base.:(==)(X::NormalToricVariety, Y::NormalToricVariety)
+  X === Y && return true
+  ambient_dim(X) == ambient_dim(Y) || return false
+  n_rays(X) == n_rays(Y) || return false
 
+  # p is a permutation such that the i-th ray of X is the p(i)-th ray of Y
+  p = inv(perm(sortperm(rays(X)))) * perm(sortperm(rays(Y)))
 
+  for i in 1:n_rays(X)
+    rays(X)[i] == rays(Y)[p(i)] || return false
+  end
+  @inline rows(Z) = [row(maximal_cones(IncidenceMatrix, Z), i) for i in 1:n_maximal_cones(Z)]
+  return Set(map(r -> Set(p.(r)), rows(X))) == Set(rows(Y))
+end
+
+# @doc raw"""
+#     hash(X::NormalToricVariety, h::UInt)
+#
+# Computes a hash of a normal toric variety `X` with the property that, outside of hash collisions, `X_1 == X_2` if and only if `hash(X_1) == hash(X_2)`.
+# """
+function Base.hash(X::NormalToricVariety, h::UInt)
+  p = inv(perm(sortperm(rays(X))))
+  @inline rows(Z) = [row(maximal_cones(IncidenceMatrix, Z), i) for i in 1:n_maximal_cones(Z)]
+  set_of_maximal_cones = Set(map(r -> Set(p.(r)), rows(X)))
+  sorted_rays = Vector.([rays(X)[p(i)] for i in 1:n_rays(X)])
```

Comment:
```suggestion
  sorted_rays = permuted(collect(rays(X)), inv(p))
```
Maybe use permuted? It would be better to keep the rays as `RayVectors` (since their hash method should ignore scaling).

Reply:
Used `permuted`, as you suggested. I kept as `Vector`, unlike you suggested, since hash of two RayVectors is often the same:

```
julia> ray_generators = [[1, 1], [1, 2]];

julia> max_cones = incidence_matrix([[1, 2]])
1×2 IncidenceMatrix
[1, 2]


julia> X = normal_toric_variety(max_cones, ray_generators)
Normal toric variety

julia> hash(rays(X)[1]) == hash(rays(X)[2])
true
```

Reply:
I do remember now, for efficiency we consider only the sign vector for the hash and allow fallback to a proper comparison otherwise. (To avoid having to normalize the vectors for the hash. This does not make as much of a difference for rational numbers but it does for more complicated fields)

The ray vectors from a polyhedral fan should be normalized so this should be fine in this case. (In general `RayVector` is not normalized)